tests/api_client/test_api_client.py

```diff
@@ -0,0 +1,309 @@
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""
+Unit tests for API client
+"""
+
+import pytest
+import json
+from strawberryfields import api_client
+from strawberryfields.api_client import (
+    requests,
+    Job,
+    ResourceManager,
+    ObjectAlreadyCreatedException,
+    MethodNotSupportedException,
+)
+
+from unittest.mock import MagicMock
+
+status_codes = requests.status_codes.codes
+
+
+@pytest.fixture
+def client():
+    return api_client.APIClient()
+
+
+SAMPLE_JOB_CREATE_RESPONSE = {
+    "id": 29583,
+    "status": "queued",
+    "result_url": "https://platform.xanadu.ai/jobs/29583/result",
+    "circuit_url": "https://platform.xanadu.ai/jobs/29583/circuit",
+    "created_at": "2019-05-24T15:55:43.872531Z",
+    "started_at": None,
+    "finished_at": None,
+    "running_time": None,
+}
+
+SAMPLE_JOB_RESPONSE = {
+    "id": 19856,
+    "status": "complete",
+    "result_url": "https://platform.xanadu.ai/jobs/19856/result",
+    "circuit_url": "https://platform.xanadu.ai/jobs/19856/circuit",
+    "created_at": "2019-05-24T15:55:43.872531Z",
+    "started_at": "2019-05-24T16:01:12.145636Z",
+    "finished_at": "2019-05-24T16:01:12.145645Z",
+    "running_time": "9µs"
+}
+
+
+class MockResponse:
+    status_code = None
+
+    def __init__(self, status_code):
+        self.status_code = status_code
+
+    def json(self):
+        return self.possible_responses[self.status_code]
+
+    @property
+    def text(self):
+        return json.dumps(self.json())
+
+    def raise_for_status(self):
+        raise requests.exceptions.HTTPError()
+
+
+class MockPOSTResponse(MockResponse):
+    possible_responses = {
+        201: SAMPLE_JOB_CREATE_RESPONSE,
+        400: {
+            "code": "parse-error",
+            "detail": (
+                "The blackbird script could not be parsed. "
+                "Please fix errors in the script and try again.")
+        },
+        401: {
+            "code": "unauthenticated",
+            "detail": "Requires authentication"
+        },
+        409: {
+            "code": "unsupported-circuit",
+            "detail": (
+                "This circuit is not compatible with the specified hardware.")
+        },
+        500: {
+            "code": "server-error",
+            "detail": (
+                "Unexpected server error. Please try your request again "
+                "later.")
+        },
+    }
+
+
+class MockGETResponse(MockResponse):
+    possible_responses = {
+        200: SAMPLE_JOB_RESPONSE,
+        401: {
+            "code": "unauthenticated",
+            "detail": "Requires authentication"
+        },
+        404: {
+            "code": "",
```

Comment:
why blank entries here?

Reply:
Forgot to put data in here, will update.